logic/atrium-logic-common/src/main/kotlin/ch/tutteli/atrium/logic/creating/basic/contains/creators/impl/ContainsObjectsAssertionCreator.kt

```diff
@@ -31,17 +33,23 @@ abstract class ContainsObjectsAssertionCreator<T : Any, TT : Any, in SC, S : Con
     checkers: List<C>
 ) : ContainsAssertionCreator<T, TT, SC, C>(searchBehaviour, checkers) {
 
-    override fun searchAndCreateAssertion(
+    final override fun searchAndCreateAssertion(
         multiConsumableContainer: AssertionContainer<TT>,
         searchCriterion: SC,
-        featureFactory: (Int, Translatable) -> AssertionGroup
+        featureFactory: (Int, Translatable) -> AssertionGroup?
     ): AssertionGroup {
         val count = search(multiConsumableContainer, searchCriterion)
         val featureAssertion = featureFactory(count, descriptionNumberOfOccurrences)
+        val mismatchedAssertions = mismatches(multiConsumableContainer, searchCriterion)
+        val assertions = if (searchBehaviour is NotSearchBehaviour) {
+            listOfNotNull(createExplanatoryGroupForMismatches(mismatchedAssertions))
```

Comment:
I see, I didn't realize assertions had to be a non-empty list. When there are no sub-assertions in the list, it seems that we would rather return a descriptive assertion: `assertionBuilder.createDescriptive(groupDescription, searchCriterion, trueProvider).build()`? However we may need to wrap the descriptive assertion in an `invisibleGroup` since the `searchAndCreateAssertion` requires an `AssertionGroup` and not an `Assertion`.

Reply:
Yes, it seems so!